### ISSUE TITLE
Dropdown

### ISSUE THREAD

**Issue by michaeltheb** [Morgul/ui-bootstrap4#7]
hello, congratulations for the work you did.
I'm working on my theme and I found a small bug about the dropdowns, if I insert a `div `element inside the button that opens the dropdown, the dropdown itself does not work, because it adds the `show `class to the internal `div `at the button, there are solutions for this bug? this is my code:

```
<li class="nav-item dropdown" uib-dropdown ng-controller="InboxCtrl" on-toggle="toggled(open)" >
	<a href="javascript:void(0)" uib-dropdown-toggle id="inbox-messages-dropdown">
		<div notify count='scopeVariable'>
			<i class="ti-comment"></i>
		</div>
		<span translate="topbar.messages.MAIN">MESSAGES</span>
	</a>
	<div class="dropdown-menu dropdown-menu-right dropdown-light dropdown-messages dropdown-large" uib-dropdown-menu aria-labelledby="inbox-messages-dropdown">
		<div class="dropdown-header" translate="topbar.messages.HEADER"> Unread messages</div>
		<div class="drop-down-wrapper ps-container">
			<ul>
				<li class="unread" ng-repeat="message in messages | orderBy: 'date':true | limitTo:3 ">
					<a href="javascript:;" ng-class="{ unread: !message.read }">
						<div class="clearfix">
							<div class="thread-image">
								<img ng-src="{{ message.avatar && message.avatar || noAvatarImg }}" alt="" class="img-responsive img-rounded">
							</div>
							<div class="thread-content">
								<span class="author">{{message.from}}</span>
								<span class="preview">{{ message.content | htmlToPlaintext | words:15:true }}</span>
								<span class="time"> {{ message.date | date: "MM/dd/yyyy 'at' h:mma" }}</span>
							</div>
						</div>
					</a>
				</li>
			</ul>
		</div>
		<div class="dropdown-divider"></div>
		<div class="view-all">
			<a href="#" translate="topbar.messages.SEEALL"> See All </a>
		</div>
	</div>
</li>
```

thank you for your support.

**Comment by Morgul** [comment]
Try using any other element than a div. (A `<span style="display: block">` for example.)

I'm not sure what might be causing this, but I feel like it's the selector logic, probably.

**Comment by michaeltheb** [comment]
Thanks for the reply,
unfortunately it is a directive that in addition to the initial div adds other div within, ie I should change the directive, which I did not but is downloaded

**Comment by Morgul** [comment]
@michaeltheb I will gladly accept a merge request on this issue, but I currently don't have the time to devote to fixing it. But, if you can put something together that fits your needs (and doesn't break the general use case), by all means, I will accept it, and push out a new version with the fix ASAP.

**Comment by YuraDubensky** [comment]
Hi, I think I found a solution for this issue and created a pull-request.
Fix is very simple - we should find dropdown-menu container by "uib-dropdown-menu" directive, not by 'div' selector.

I also found one more compatibility issue - latest boostrap sets 'show' class on both dropdown-menu and parent element.
https://getbootstrap.com/docs/4.0/components/dropdowns/
So I also added logic to add openClass to uib-dropdown element in pull-request.

I have tested my changes on my env - works just fine.

Please review and merge if it looks ok to you, so I can update with new version on my project.

PS: Thanks for developing such a great lib!

**Comment by Morgul** [comment]
closed by 6cda57a9b9cf083f14b161c49632c971e5be10bd
